components/eam/src/chemistry/yaml/modal_aero_wateruptake/f90_yaml/wateruptake_dryaer_beg_yml.f90

```diff
@@ -0,0 +1,81 @@
+  integer  :: unit_input, unit_output, y_nstep
+  integer,save :: n_calls=0   ! some subroutines are called multiple times in one timestep, record the number of calls
+
+  !populate YAML structure
+  yaml%lev_print = 51 !level (**remove these if generating data for a dependent subroutines**)
+  yaml%nstep_print = 355 !time step(**remove these if generating data for a dependent subroutines**)
+
+  ! set ext_str if there are multiple sets of yaml output to be write out
+  ! here gives an example that "flag" in the code can be 0, 1, or 2:
+  ! write(ext_str,'(I2)') flag
+  ! ext_str = 'flag_'//adjustl(ext_str)
+
+
+  !YAML file input generation code- DO NOT PORT to C++
+  !print all inputs one-by-one at column "yaml%col_print"
+  yaml%col_print = icolprnt(y_lchnk) !column to write data(**remove these if generating data for a dependent subroutines**)
+  y_nstep = get_nstep() !time step (**remove these if generating data for a dependent subroutines**)
+
+  yaml%flag_print = .false. ! to write or not to write data (**remove these if generating data for a dependent subroutines**)
+!  if(yaml%col_print == y_i .and. y_nstep==yaml%nstep_print .and. y_k == yaml%lev_print) then ! if this column exists in y_lchnk
+     !<
+     !In the case of y_i or y_k are not passed as arguments:
+     if(yaml%col_print >0 .and. y_nstep==yaml%nstep_print) then
+
+     !For generating data for a dependent subroutines where "yaml" derived type is already initialized:
+     !if(yaml%flag_print) then
+     !>
+
+     ! record number of calls that can output yaml file. you only need to write one set of input/output
+     n_calls = n_calls+1
+     if (n_calls==1) then ! output the first call. change this if writing out other calls
+     ! if ((n_calls==1 .and. flag==0) .or. (n_calls==3 .and. flag==1) .or. (n_calls==5 .and. flag==2)) then ! this is an example of writing multiple sets of YAML files
+
+        yaml%lchnk_print = y_lchnk !(**remove these if generating data for a dependent subroutines**)
+        yaml%flag_print  = .true.!(**remove these if generating data for a dependent subroutines**)
+
+        !open I/O yaml files (it can have an extra optional argument to pass a unique string to differentiate file names)
+        call open_files('modal_aero_wateruptake_dryaer', &  !intent-in
+             unit_input, unit_output) !intent-out
+!             unit_input, unit_output, trim(ext_str)) !intent-out, with the use of ext_str
+
+
+        !start by adding an input string
+        call write_input_output_header(unit_input, unit_output,yaml%lchnk_print,yaml%col_print, &
+             'compute_tendencies',yaml%nstep_print, yaml%lev_print)
+
+        !< add code for writing data here>
+        call write_var(unit_input, unit_output, 'ncol', ncol)
+        call write_1d_var(unit_input, unit_output, 'dgncur_a', nmodes,dgncur_a(yaml%col_print, yaml%lev_print,:)) 
+        call write_1d_var(unit_input, unit_output, 'state_q', nmodes, state_q(yaml%col_print, yaml%lev_print,:))
+
```

Comment:
@tangshuaiqi I'm adding state_q as 1D input and looking for something similar. I wonder if the dimension of state_q should be pcnst? I think it contains a lot of species.

Reply:
yep... you are right. I will correct the dimension.